packet/packet.py

```diff
@@ -74,25 +137,40 @@ def get_current_packet(freshman_username):
     return get_freshman(freshman_username).current_packet()
 
 
-@lru_cache(maxsize=2048)
-def get_number_signed(freshman_username):
-    return get_current_packet(freshman_username).signatures_received()
 
 
 @lru_cache(maxsize=2048)
-def get_number_required(freshman_username):
-    return get_current_packet(freshman_username).signatures_required()
+def get_number_signed(freshman_username, separated=False):
+    """
+    Gets the raw number of signatures for the user
+    :param freshman_username: The user to get signature numbers for
+    :param separated: A boolean indicating whether to return the results as separated
+    :return: <Packet> list of results that are in the form of Packet database objects
+    """
+    return db.session.query(Packet) \
+        .filter(Packet.freshman_username == freshman_username,
+                Packet.start < datetime.now(), Packet.end > datetime.now()) \
+        .first().signatures_received(not separated)
+
+
+@lru_cache(maxsize=4096)
+def get_number_required(separated=False):
+    """
+    Get the number of required signatures for Packet (not counting on/off-floor status)
+    :param separated: whether or not to separate those by category
+    :return: a map or an integer of total signatures required
+    """
+    return db.session.query(Packet) \
+        .filter(Packet.start < datetime.now(), Packet.end > datetime.now()).first().signatures_required(not separated)
 
 
 @lru_cache(maxsize=2048)
-def get_upperclassmen_percent(uid):
-    upperclassmen_required = copy.deepcopy(get_number_required(uid))
-    del upperclassmen_required['freshmen']
-    upperclassmen_required = sum(upperclassmen_required.values())
-
-    upperclassmen_signature = copy.deepcopy(get_number_signed(uid))
-    del upperclassmen_signature['freshmen']
-    upperclassmen_signature = sum(upperclassmen_signature.values())
+def get_upperclassmen_percent(username, onfloor=False):
```

Comment:
Does this work without the deep copy? I was getting a lot of issues without the deep copy

Reply:
It was actually calculating wrong, I fixed it still without deep copy.  The query itself on the get_signed in model.py was changed in this merge so it works without it